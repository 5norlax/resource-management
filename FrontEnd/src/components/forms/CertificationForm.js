import React from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';

const CertificationForm = ({ index, removeCertification, CertName, CertDate, CertAssociation, handleCertificationChange }) => {
    var today = new Date();
    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="CertName">Name</Label>
                    <Input type="text" name="CertName" id="CertName" placeholder="Certification Name" value={CertName} onChange={(e) => { handleCertificationChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="CertDate">Date</Label>
                    <Input type="number" name="CertDate" id="CertDate" min="1900" max={today.getFullYear()} step="1" placeholder="Certification Date" value={CertDate} onChange={(e) => { handleCertificationChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="CertAssociation">Association</Label>
                    <Input type="text" name="CertAssociation" id="CertAssociation" placeholder="Certification Association" value={CertAssociation} onChange={(e) => { handleCertificationChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 text-right">
                    <Button onClick={() => { removeCertification(index) }} style={{ marginTop: '32px' }} color="danger">Remove Education</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default CertificationForm;