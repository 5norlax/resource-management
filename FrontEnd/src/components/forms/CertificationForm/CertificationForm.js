import React from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';

export default function CertificationForm({ index, removeCertification, CertName, CertDate, CertAssociation, handleCertificationChange }) {

    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="CertName">Name</Label>
                    <Input required type="text" name="CertName" id="CertName" placeholder="Certification Name" value={CertName} onChange={(e) => { handleCertificationChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="GradDate">Date Achieved</Label>
                    <Input
                        required
                        type="date"
                        name="CertDate"
                        id="CertDate"
                        value={CertDate}
                        placeholder="date placeholder"
                        onChange={(e) => { handleCertificationChange(index, e) }}
                    />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="CertAssociation">Association</Label>
                    <Input required type="text" name="CertAssociation" id="CertAssociation" placeholder="Certification Association" value={CertAssociation} onChange={(e) => { handleCertificationChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 text-right">
                    <Button onClick={() => { removeCertification(index) }} style={{ marginTop: '32px' }} color="danger">Remove Certification</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
};