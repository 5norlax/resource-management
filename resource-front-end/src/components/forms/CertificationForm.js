import React from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';

const CertificationForm = ({index, removeCertification}) => {
    return (
        <div>
            <Row>
            <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="certName">Name</Label>
                    <Input type="text" name="certName" id="certName" placeholder="Certification Name" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="certDate">Date</Label>
                    <Input type="text" name="certDate" id="certDate" placeholder="Date Completed" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="certAssociation">Association</Label>
                    <Input type="text" name="certAssociation" id="certAssociation" placeholder="Certification Association" />
                </FormGroup>
                <FormGroup className="col-xl-2 col-lg-2 col-md-2 col-sm-3 col-xs-3">
                    <Button onClick={() => {removeCertification(index)}} style={{position: 'absolute', bottom: 0, right: 0}} color="danger">Remove Education</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default CertificationForm;