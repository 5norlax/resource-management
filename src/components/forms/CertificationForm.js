import React from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';

const CertificationForm = ({index, removeCertification}) => {
    return (
        <div>
            <Row>
            <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="certName">Name</Label>
                    <Input type="text" name="certName" id="certName" placeholder="Certification Name" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="certDate">Date</Label>
                    <Input type="text" name="certDate" id="certDate" placeholder="Date Completed" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="certAssociation">Association</Label>
                    <Input type="text" name="certAssociation" id="certAssociation" placeholder="Certification Association" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 text-right">
                    <Button onClick={() => {removeCertification(index)}} style={{ marginTop: '32px' }} color="danger">Remove Education</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default CertificationForm;