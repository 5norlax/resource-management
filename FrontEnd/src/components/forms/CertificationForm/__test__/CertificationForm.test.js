import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CertificationForm from '../CertificationForm';

afterEach(cleanup);

it('matches snapshot', async () => {
  const tree = renderer.create(<CertificationForm />).toJSON()
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', async () => {
  const div = document.createElement('div');
  ReactDOM.render(<CertificationForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Input Fields', () => {
  const handleCertificationChange = jest.fn();
  it('Name Input triggers handleCertificationChange onChange', async () => {
    const {queryByPlaceholderText} = render(<CertificationForm handleCertificationChange={handleCertificationChange} />);
    const nameInput = queryByPlaceholderText('Certification Name');
    fireEvent.change(nameInput, {target: {value: "test"}});
    expect(handleCertificationChange).toHaveBeenCalled();
  });
  it('Date Input triggers handleCertificationChange onChange', async () => {
    const {queryByTestId} = render(<CertificationForm handleCertificationChange={handleCertificationChange} />);
    const dateInput = queryByTestId('certDateInput');
    fireEvent.change(dateInput, {target: {value: "0001-01-01"}});
    expect(handleCertificationChange).toHaveBeenCalled();
  });
  it('Association Input triggers handleCertificationChange onChange', async () => {
    const {queryByPlaceholderText} = render(<CertificationForm handleCertificationChange={handleCertificationChange} />);
    const associationInput = queryByPlaceholderText('Certification Association');
    fireEvent.change(associationInput, {target: {value: "test"}});
    expect(handleCertificationChange).toHaveBeenCalled();
  });
});

describe('Remove Button', () => {
  it('Button Removes Certification onClick', async () => {
    const removeCertification = jest.fn();
    const {queryByText} = render(<CertificationForm removeCertification={removeCertification} />);
    fireEvent.click(queryByText('Remove Certification'));
    expect(removeCertification).toHaveBeenCalled();
  });
});