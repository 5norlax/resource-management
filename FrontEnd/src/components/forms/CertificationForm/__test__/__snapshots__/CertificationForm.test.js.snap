// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="CertName"
      >
        Name
      </label>
      <input
        className="form-control"
        id="CertName"
        name="CertName"
        onChange={[Function]}
        placeholder="Certification Name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="GradDate"
      >
        Date Achieved
      </label>
      <input
        className="form-control"
        data-testid="certDateInput"
        id="CertDate"
        name="CertDate"
        onChange={[Function]}
        placeholder="date placeholder"
        required={true}
        type="date"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="CertAssociation"
      >
        Association
      </label>
      <input
        className="form-control"
        id="CertAssociation"
        name="CertAssociation"
        onChange={[Function]}
        placeholder="Certification Association"
        required={true}
        type="text"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 text-right form-group"
    >
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        style={
          Object {
            "marginTop": "32px",
          }
        }
        type="button"
      >
        Remove Certification
      </button>
    </div>
  </div>
  <hr />
</div>
`;
